C/plugins/south/ina219/plugin.cpp

```diff
@@ -0,0 +1,146 @@
+/*
+ * FogLAMP south plugin.
+ *
+ * Copyright (c) 2018 Dianomic Systems
+ *
+ * Released under the Apache 2.0 Licence
+ *
+ * Author: Mark Riddoch
+ */
+#include <ina219.h>
+#include <plugin_api.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <strings.h>
+#include <string>
+#include <logger.h>
+#include <plugin_exception.h>
+#include <config_category.h>
+#include <rapidjson/document.h>
+
+using namespace std;
+
+/**
+ * Default configuration
+ */
+#define CONFIG	"{\"plugin\" : { \"description\" : \"INA219 current and voltage sensor\", " \
+			"\"type\" : \"string\", \"default\" : \"ina219\" }, " \
+		"\"asset\" : { \"description\" : \"Asset name\", "\
+			"\"type\" : \"string\", \"default\" : \"electrical\" }, " \
+		"\"address\" : { \"description\" : \"Address of IAN219\", " \
+			"\"type\" : \"integer\", \"default\" : \"64\" }, "\
+		"\"range\" : { \"description\" : \"Required range setting\", " \
+			"\"type\" : \"string\", \"default\" : \"32V2A\" } "\
+			"}"
+
+/**
+ * The INA219 plugin interface
+ */
+extern "C" {
+
+/**
+ * The plugin information structure
+ */
+static PLUGIN_INFORMATION info = {
+	"ina219",                 // Name
+	"1.0.0",                  // Version
+	0,    			  // Flags
+	PLUGIN_TYPE_SOUTH,        // Type
+	"1.0.0",                  // Interface version
+	CONFIG			  // Default configuration
+};
+
+/**
+ * Return the information about this plugin
+ */
+PLUGIN_INFORMATION *plugin_info()
+{
+	return &info;
+}
+
+/**
+ * Initialise the plugin, called to get the plugin handle
+ */
+PLUGIN_HANDLE plugin_init(ConfigCategory *config)
+{
+INA219 *ina219 = 0;
+string	address, range;
+
+	if (config->itemExists("address"))
+	{
+		address = config->getValue("address");
+		ina219 = new INA219(atoi(address.c_str()));
```

Comment:
May be use std::stoi rather than atoi. "atoi" behaves badly for bad/invalid inputs.